src/Core/Settings.cpp

```diff
@@ -2302,6 +2302,9 @@ What to do when the limit is exceeded.
 )", 0) \
     DECLARE(UInt64, max_bytes_before_external_group_by, 0, R"(
 If memory usage during GROUP BY operation is exceeding this threshold in bytes, activate the 'external aggregation' mode (spill data to disk). Recommended value is half of the available system memory.
+)", 0) \
+    DECLARE(Double, max_bytes_ratio_before_external_group_by, 0., R"(
```

Comment:
Requires adding it to the documentation.

Reply:
Done [b72e705](https://github.com/ClickHouse/ClickHouse/pull/71406/commits/b72e70576781667b0cc9d326f4cbd33584bbf041)